include/swift/Reflection/ReflectionContext.h

```diff
@@ -801,7 +801,50 @@ class ReflectionContext
     }
 
     case RecordKind::MultiPayloadEnum: {
```

Comment:
This block of code actually identifies the current case for multi-payload enums.  Today, it only handles enums that use a separate tag (as inferred from the size data exposed in the type information).  For other enum layouts it returns `false` to indicate that it is unable to parse that particular enum.